### PR TITLE
build: update Dockerfile

### PR DESCRIPTION
**Pull request by jimlee2002**
- use base image `scratch` for super minimal image
- support ssl
- allow env `TZ=` to work with possible builtin cron setting in future
- use exec form `ENTRYPOINT` for defaults & safety exec
- update README.md, add toc

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,27 @@
...
+    /tmp/reaper \
+    /tmp/reaper
+
 COPY reaper /usr/local/bin/reaper
 
-CMD ["reaper"]
\ No newline at end of file
+WORKDIR /reaper
+
+ENTRYPOINT ["reaper","-c","/reaper/config.yaml"]
```
Comment (LeslieLeung) [inline Dockerfile:26]: Should remove the "-c" parameter as it is already default. See [here](https://github.com/LeslieLeung/reaper/blob/0214f53463a18fd8bad8580bb4d881d1b4371985/cmd/root.go#L29C1-L29C1).

Comment (jimlee2002) [reply Dockerfile]: current defautlt config path used in docker is `/reaper/config.yaml`, which is different with the default set in program. 
maybe set the same defautlt config path in docker container?

Comment (jimlee2002) [reply Dockerfile]: > current defautlt config path used in docker is `/reaper/config.yaml`, which is different with the default set in program. maybe set the same default config path in docker container?

Updated.

**Review by LeslieLeung — APPROVED** [review]
Thank you for the contribution.

**Pull request merged**